app/src/main/java/mozilla/lockbox/view/ItemDetailFragment.kt

```diff
@@ -90,8 +90,17 @@ class ItemDetailFragment : BackableFragment(), ItemDetailView {
         inputLayoutPassword.isHintAnimationEnabled = false
 
         inputUsername.readOnly = true
-        inputUsername.isClickable = true
-        inputUsername.isFocusable = true
+
+        if (item.username.isNullOrEmpty()) {
```

Comment:
`if / else` in your `Fragment` is a code smell -- I would rather have two methods on the `ItemListView` interface that do something like `showUsernamePlaceholder()` and `showUsername(username: String)` (for the `if` and `else` cases, respectively). The `Presenter` can drive those, making this behavior unit testable.

Reply:
👏 this would make things so much easier, because I was unable to access the strings because the fragment doesn't have any context to find the `getString` methods.